python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -723,19 +726,42 @@ def should_reconcile(
             return False
 
         if all(parents_will_be_reconciled(asset_graph, candidate) for candidate in candidates_unit):
-            unit_materialize_reason = next(
+            unit_condition = next(
                 filter(
                     None,
-                    (reason_for_candidate(candidate) for candidate in candidates_unit),
+                    (condition_for_candidate(candidate) for candidate in candidates_unit),
                 ),
                 None,
             )
-            if unit_materialize_reason:
+            if unit_condition:
                 for candidate in candidates_unit:
-                    conditions_by_asset_partition[candidate].add(unit_materialize_reason)
-                return (
-                    unit_materialize_reason.decision_type == AutoMaterializeDecisionType.MATERIALIZE
-                )
+                    conditions_by_asset_partition[candidate].add(unit_condition)
+                # if the condition indicates we're meant to materialize this candidate unit, we
+                # must first check if the rate limit has been exceeded
+                if unit_condition.decision_type == AutoMaterializeDecisionType.MATERIALIZE:
+                    # if any candidate in the unit exceeds its limit...
+                    max_materializations_exceeded = False
+                    for candidate in candidates_unit:
+                        materialization_counts_by_asset_key[candidate.asset_key] += 1
+                        auto_materialize_policy = get_implicit_auto_materialize_policy(
+                            asset_graph=asset_graph, asset_key=candidate.asset_key
+                        )
+                        if (
+                            auto_materialize_policy
+                            and auto_materialize_policy.max_materializations_per_minute
+                            and materialization_counts_by_asset_key[candidate.asset_key]
+                            > auto_materialize_policy.max_materializations_per_minute
+                        ):
+                            max_materializations_exceeded = True
+                            break
+                    # ... then all assets in the unit should have the limit exceeded condition
+                    if max_materializations_exceeded:
+                        for candidate in candidates_unit:
```

Comment:
Feel free to decide this is aesthetically worse,  but all this logic could go inside the `if` statement above, right? And then just `return False` instead of breaking?